src/main/java/domain/Rank.java

```diff
@@ -0,0 +1,36 @@
+package domain;
+
+import java.util.Arrays;
+
+public enum Rank {
+    NONE(0, 0),
+
+    FIFTH(3, 5_000),
+    FOURTH(4, 50_000),
+    THIRD(5, 1_500_000),
+    SECOND(5, 30_000_000),
+    FIRST(6, 2_000_000_000);
+
+    private int count;
+    private int prize;
+
+    Rank(int count, int prize) {
+        this.count = count;
+        this.prize = prize;
+    }
+
+    public static Rank matchRank(int count, boolean isMatchBonus) {
+        if (count == SECOND.count && isMatchBonus) {
+            return SECOND;
+        }
+
+        return Arrays.stream(values())
+                .filter(rating -> rating.count == count)
+                .findFirst()
+                .orElse(NONE);
+    }
+
+    public static long calculateTotalPrize(Rank rank, Integer count) {
+        return (long) rank.prize * count;
+    }
```

Comment:
static 메서드로 만들필요가 없이 인스턴스의 메서드로 만들어도 충분할 것 같습니다. 😃

```suggestion
    public long calculateTotalPrize(Integer count) {
        return (long) this.prize * count;
    }
```